lib/kubernetes-deploy/kubernetes_resource/redis.rb

```diff
@@ -41,6 +43,15 @@ def service_ready?
       @service.dig("spec", "clusterIP").present?
     end
 
+    def name
+      return unless @instance_data.present?
```

Comment:
Since the name couldn't possibly have changed between the live copy (`@instance_data`) and the in-repo copy (`@definition`), I'd suggest using `@definition` for this instead.